pkg/proxyagent/agent/agent.go

```diff
@@ -73,6 +74,8 @@ func NewAgentAddon(
 		},
 	}
 	// Register the custom signer CSR option if V1 csr is supported
+	// caculate a hash value of signer ca data and add it to the organizationUnits of the subject
+	signerHash := sha256.Sum256(signer.CAData())
```

Comment:
can we just use service endpoint?

Reply:
service namespace and name is configurable in cluster-proxy: https://github.com/xuezhaojun/cluster-proxy/blob/8c762cca2bbc65e4fa5ecce0d5e8ec264fe4e3bd/pkg/proxyserver/controllers/manifests.go#L56-L57

Reply:
i mean the endpoint exposed publicly, not internally

Reply:
like route url in ocp case?

Reply:
I think we need that as CN to sign the serving CA, isn't it?

Reply:
I think you mean https://github.com/stolostron/cluster-proxy/blob/d60d4b3328e44114c283da343a3ef2638c2e0eed/charts/cluster-proxy/templates/managedproxyconfiguration.yaml#L15-L16

this entrypoint has multiple types, is is used as SANs when sign: https://github.com/stolostron/cluster-proxy/blob/d60d4b3328e44114c283da343a3ef2638c2e0eed/pkg/proxyserver/controllers/managedproxyconfiguration_controller.go#L388

But since it is in managedproxyconfiguration, there is chance the entrypoint change to another type.

Reply:
If it is changed, do we need to re-sign the serving cert, and re-trigger the CSR?

Reply:
No need to re-trigger CSR, the agent deployment will be updated to access the new endpoint but it can use the previous signed client cert.

The server certificates will be rotate by the managedproxyconfiguraion_controller on the hub side.